gateway-ha/src/test/java/io/trino/gateway/proxyserver/TestProxyRequestHandler.java

```diff
@@ -77,6 +77,14 @@ public MockResponse dispatch(RecordedRequest request)
                             .setBody("{\"starting\": false}");
                 }
 
+                if (request.getPath().equals(healthCheckEndpoint + "?test-compression")) {
```

Comment:
Please squash the 3rd commit into 2nd commit. We don't separate a commit for tests.